shared/airgap.go

```diff
@@ -0,0 +1,180 @@
+package shared
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/rancher/distros-test-framework/pkg/customflag"
+
+	. "github.com/onsi/gomega"
```

Comment:
so this pkg we are using to shared base "utils" ,helpers etc.... on golang its one of the most best practices is error handling , always returning the error to the layer above ( who calls your helper method ) , so that guy will have the ability and flexibility to do what he wants if the error returned ( which in a lot of possible scnearios you don wanna to kill execution and handle it in a different or graceful way )

saying that , using gomega ( one of the reasons i dont like it ) you have no way of doing that because of the `Expect(err).To(BeNil())`


if you noticed anywhere on this pkg we have used gomega because of that, to maintain the ability of doing that 

i would strongly recommend to wherever needs, have a err check and return it appropriately not using gomega.

and for the function witgh go routines you will need to create a chan of errors, have the  functions 
```
copyAssets(cluster, nodeIP)
			copyRegistry(cluster, nodeIP)
			makeExecs(cluster, nodeIP)
```

return if any error to this channel, in the end, loop. in the channel to gather errros and  close the channel 
or if any of those funcs you need to fail fast if has an errors , just close the chan right after, instead of first gather errors there , then loop, then close.

if you want i can help with that tomorrow or thursday, like give me some functions i refactor and sent you back. to not give you too much trouble.

let me know

Reply:
Sure agree with returning errors, the reason for adding Except is to fail the test of any of the test fails as if any of these steps fail the test should fail as well... I can return the error on Except
For the channels, I can try and implement that and can ping you if I need any help with it...